### PR TITLE
feat(api): set common logger slog

### PR DESCRIPTION
**Pull request by Isteb4k**
## Description

Refactored controller to single logger interface (slog)

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

### CODE REVIEW

File: images/virtualization-artifact/pkg/controller/cvi/cvi_reconciler.go [hunk 1]
```diff
@@ -66,18 +69,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
-	var requeue bool
+	log.Info("Start cvi reconciliation")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/cvi/cvi_reconciler.go:72]: Do we always want to know about the beginning of the reconciliation? Seems redundant for 'INFO'

File: images/virtualization-artifact/pkg/controller/cvi/cvi_reconciler.go [hunk 2]
```diff
@@ -100,14 +102,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
-		slog.Info("Requeue")
+		log.Info("Requeue cvi reconciliation")
 		return reconcile.Result{
 			RequeueAfter: 2 * time.Second,
 		}, nil
 	}
 
-	slog.Info("Done")
-
+	log.Info("Finished cvi reconciliation")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/cvi/cvi_reconciler.go:111]: Seems redundant for 'INFO'

File: images/virtualization-artifact/pkg/controller/service/ip_address_service.go [hunk 3]
```diff
@@ -33,15 +33,15 @@ type IpAddressService struct {
...
-	logger logr.Logger,
+	logger *slog.Logger,
 	virtualMachineCIDRs []string,
 ) *IpAddressService {
 	parsedCIDRs := make([]netip.Prefix, len(virtualMachineCIDRs))
 
 	for i, cidr := range virtualMachineCIDRs {
 		parsedCIDR, err := netip.ParsePrefix(cidr)
 		if err != nil {
-			logger.Error(err, fmt.Sprintf("failed to parse CIDR %s:", cidr), "err", err)
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/service/ip_address_service.go:44]: Md I'm wrong, but there must be mistakes, that the 'Error' should be with a small letter

Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/service/ip_address_service.go:44]: It's not error, just string

File: images/virtualization-artifact/pkg/controller/vmiplease/vmiplease_reconciler.go [hunk 4]
```diff
@@ -106,17 +106,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
-	r.logger.Info("Start reconcile VMIPLease", "namespacedName", req.String())
+	log.Info("Start reconcile VMIPLease")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmiplease/vmiplease_reconciler.go:109]: I would return the namespace and put it in DEBUG

File: images/virtualization-artifact/pkg/controller/vmip/vmip_reconciler.go [hunk 5]
```diff
@@ -143,17 +143,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
-	r.logger.Info("Start reconcile VMIP", "namespacedName", req.String())
+	log.Info("Start reconcile VMIP")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmip/vmip_reconciler.go:146]: I would return the namespacedName and put it in DEBUG

File: images/virtualization-artifact/pkg/controller/vmip/internal/iplease_handler.go [hunk 6]
```diff
@@ -129,7 +131,7 @@ func (h IPLeaseHandler) createNewLease(ctx context.Context, state state.VMIPStat
 	if err != nil {
 		vmipStatus.Address = ""
 		msg := fmt.Sprintf("the VirtualMachineIP cannot be created: %s", err.Error())
-		h.logger.Info(msg)
+		log.Info(msg)
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmip/internal/iplease_handler.go:134]: Let's translate this into 'Error' as part of your PR

File: images/virtualization-artifact/pkg/controller/vmclass/vmclass_reconciler.go [hunk 7]
```diff
@@ -99,38 +99,41 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
 
 	if class.IsEmpty() {
-		r.logger.Info("Reconcile observe an absent VirtualMachineClass: it may be deleted", slog.String("namespacedName", req.String()))
+		log.Info("Reconcile observe an absent VirtualMachineClass: it may be deleted")
 		return reconcile.Result{}, nil
 	}
 	s := state.New(r.client, class)
 
-	r.logger.Info("Start reconcile VMClass", slog.String("namespacedName", req.String()))
+	log.Info("Start reconcile VMClass")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmclass/vmclass_reconciler.go:107]: I would return the namespacedName and put it in DEBUG

File: images/virtualization-artifact/pkg/controller/vmclass/vmclass_reconciler.go [hunk 8]
```diff
@@ -99,38 +99,41 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
 	}
 	err = class.Update(ctx)
 	if err != nil {
-		r.logger.Error("Failed to update VirtualMachineClass", slog.String("namespacedName", req.String()))
+		log.Error("Failed to update VirtualMachineClass")
 		return reconcile.Result{}, err
 	}
-	r.logger.Info("Finished reconcile VM", slog.String("namespacedName", req.String()))
+
+	log.Info("Finished reconcile VM")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmclass/vmclass_reconciler.go:136]: I would return the namespacedName and put it in DEBUG

File: images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vm_watcher.go [hunk 9]
```diff
@@ -66,7 +64,7 @@ func (w VirtualMachineWatcher) enqueueRequests(ctx context.Context, obj client.O
 		Namespace: obj.GetNamespace(),
 	})
 	if err != nil {
-		w.logger.Error(fmt.Sprintf("failed to list vmbdas: %s", err))
+		slog.Default().Error(fmt.Sprintf("failed to list vmbdas: %s", err))
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vm_watcher.go:67]: Why use default logger in this case?

File: images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vd_watcher.go [hunk 10]
```diff
@@ -66,7 +64,7 @@ func (w VirtualDiskWatcher) enqueueRequests(ctx context.Context, obj client.Obje
 		Namespace: obj.GetNamespace(),
 	})
 	if err != nil {
-		w.logger.Error(fmt.Sprintf("failed to list vmbdas: %s", err))
+		slog.Default().Error(fmt.Sprintf("failed to list vmbdas: %s", err))
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vd_watcher.go:67]: Why use default logger in this case?

File: images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/kvvmi_watcher.go [hunk 11]
```diff
@@ -127,7 +123,7 @@ func (eh KVVMIEventHandler) getHotPluggedVolumeStatuses(obj client.Object) map[s
 			var name string
 			name, ok = kvbuilder.GetOriginalDiskName(vs.Name)
 			if !ok {
-				eh.logger.Warn("VolumeStatus is not a Disk", "vsName", vs.Name, "name", kvvmi.Name, "ns", kvvmi.Namespace)
+				slog.Default().Warn("VolumeStatus is not a Disk", "vsName", vs.Name, "name", kvvmi.Name, "ns", kvvmi.Namespace)
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/kvvmi_watcher.go:126]: Why usr default logger in this case? A common question about the watcher package

File: images/virtualization-artifact/pkg/controller/vmbda/internal/life_cycle.go [hunk 12]
```diff
@@ -20,32 +20,30 @@ import (
...
 		attacher: attacher,
 	}
 }
 
 func (h LifeCycleHandler) Handle(ctx context.Context, vmbda *virtv2.VirtualMachineBlockDeviceAttachment) (reconcile.Result, error) {
-	logger := h.logger.With("name", vmbda.Name, "ns", vmbda.Namespace)
-	logger.Info("Sync")
+	log := logger.FromContext(ctx).With(logger.SlogHandler("lifecycle"))
+
 	// TODO protect vd.
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vmbda/internal/life_cycle.go:47]: Do we have a card for this revision?

File: images/virtualization-artifact/pkg/controller/vm/vm_reconciler.go [hunk 13]
```diff
@@ -302,38 +298,44 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
 	if vm.IsEmpty() {
-		r.logger.Info("Reconcile observe an absent VirtualMachine: it may be deleted", slog.String("namespacedName", req.String()))
+		log.Info("Reconcile observe an absent VirtualMachine: it may be deleted")
 		return reconcile.Result{}, nil
 	}
+
 	s := state.New(r.client, vm)
 
-	r.logger.Info("Start reconcile VM", slog.String("namespacedName", req.String()))
+	log.Info("Start reconcile VM")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vm/vm_reconciler.go:307]: I would return the namespacedName and put it in DEBUG

File: images/virtualization-artifact/pkg/controller/vm/vm_reconciler.go [hunk 14]
```diff
@@ -302,38 +298,44 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
...
+
 	err = r.updateVM(ctx, vm)
 	if err != nil {
-		r.logger.Error("Failed to update VirtualMachine", slog.String("namespacedName", req.String()))
+		log.Error("Failed to update VirtualMachine")
 		return reconcile.Result{}, err
 	}
-	r.logger.Info("Finished reconcile VM", slog.String("namespacedName", req.String()))
+
+	log.Info("Finished reconcile VM")
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vm/vm_reconciler.go:338]: I would return the namespacedName and put it in DEBUG

File: images/virtualization-artifact/pkg/controller/vm/vm_controller.go [hunk 15]
```diff
@@ -44,29 +45,30 @@ func NewController(
 	log *slog.Logger,
 	dvcrSettings *dvcr.Settings,
 ) (controller.Controller, error) {
-	if log == nil {
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vm/vm_controller.go:47]: Please, tell me why this check is no longer relevant?

File: images/virtualization-artifact/pkg/controller/vm/internal/sync_kvvm.go [hunk 16]
```diff
@@ -285,12 +291,12 @@ func (h *SyncKvvmHandler) updateKVVM(ctx context.Context, s state.VirtualMachine
...
 
-	if err := h.client.Update(ctx, kvvm); err != nil {
+	if err = h.client.Update(ctx, kvvm); err != nil {
 		return fmt.Errorf("unable to create KubeVirt VM '%s': %w", kvvm.GetName(), err)
 	}
 
-	h.logger.Info("Update KubeVirt VM done", "name", kvvm.Name)
-	h.logger.Debug("Update KubeVirt VM done", "name", kvvm.Name, "kvvm", kvvm)
+	log.Info("Update KubeVirt VM done", "name", kvvm.Name)
+	log.Debug("Update KubeVirt VM done", "name", kvvm.Name, "kvvm", kvvm)
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vm/internal/sync_kvvm.go:299]: It looks strange. Are we sure we want to leave it like this?

File: images/virtualization-artifact/pkg/controller/vm/vm_reconciler.go [hunk 17]
```diff
@@ -302,38 +298,44 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 
 	if vm.IsEmpty() {
-		r.logger.Info("Reconcile observe an absent VirtualMachine: it may be deleted", slog.String("namespacedName", req.String()))
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/pkg/controller/vm/vm_reconciler.go:305]: Mb return namespacedName?

File: images/virtualization-artifact/cmd/virtualization-controller/main.go [hunk 18]
```diff
@@ -87,7 +84,6 @@ func main() {
 
 	var pprofBindAddr string
 	flag.StringVar(&pprofBindAddr, "pprof-bind-address", os.Getenv(pprofBindAddrEnv), "enable pprof")
-	flag.Parse()
```
Comment (LopatinDmitr) [inline images/virtualization-artifact/cmd/virtualization-controller/main.go:90]: Why delete this line?

File: images/virtualization-artifact/pkg/controller/vmip/internal/iplease_handler.go (hunk 6 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vmip/internal/iplease_handler.go:134]: This is not internal error to put it to Error level

File: images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vm_watcher.go (hunk 9 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vm_watcher.go:67]: watcher doesn't have ctx to extract logger from, it's easier to use default logger here

File: images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vd_watcher.go (hunk 10 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/vd_watcher.go:67]: watcher doesn't have ctx to extract logger from, it's easier to use default logger here

File: images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/kvvmi_watcher.go (hunk 11 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vmbda/internal/watcher/kvvmi_watcher.go:126]: watcher doesn't have ctx to extract logger from, it's easier to use default logger here

File: images/virtualization-artifact/pkg/controller/vmbda/internal/life_cycle.go (hunk 12 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vmbda/internal/life_cycle.go:47]: Now we have

File: images/virtualization-artifact/pkg/controller/vm/vm_controller.go (hunk 15 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vm/vm_controller.go:47]: I found this check redundant, there is no correct case when logger is nil

File: images/virtualization-artifact/pkg/controller/vm/internal/sync_kvvm.go (hunk 16 above)
Comment (Isteb4k) [reply images/virtualization-artifact/pkg/controller/vm/internal/sync_kvvm.go:299]: Why not?

**Review by LopatinDmitr — APPROVED** [review]

**Pull request merged**